application/src/main/java/org/togetherjava/tjbot/commands/basic/VcActivityCommand.java

```diff
@@ -235,48 +232,18 @@ private static void handleErrors(@NotNull SlashCommandEvent event,
 
     /**
      * This grabs the OptionMapping, after this it <br />
-     * - validates whenever it's within an {@link Integer Integer's} range <br />
-     * - validates whenever it's positive <br />
+     * - validates whenever It's null or not <br />
      *
      * <p>
      * <p/>
      *
-     * @param event the {@link SlashCommandEvent}
-     * @param optionMapping the {@link OptionMapping}
-     * @return nullable {@link Integer}
-     * @throws java.lang.IllegalArgumentException if the option's value is - outside of
-     *         {@link Integer#MAX_VALUE} - negative
+     * @param optionMapping the {@link OptionMapping} to validate
      */
-    @Contract("_, null -> null")
-    private static @Nullable Integer handleIntegerTypeOption(@NotNull SlashCommandEvent event,
-            @Nullable OptionMapping optionMapping) {
-
-        int optionValue;
-
-        if (optionMapping == null) {
-            return null;
-        }
-
-        try {
-            optionValue = Math.toIntExact(optionMapping.getAsLong());
-        } catch (ArithmeticException e) {
-            event
-                .reply("The " + optionMapping.getName() + " is above `" + Integer.MAX_VALUE
-                        + "`, which is too high")
-                .setEphemeral(true)
-                .queue();
-            throw new IllegalArgumentException(
-                    optionMapping.getName() + " can't be above " + Integer.MAX_VALUE);
-        }
-
-        if (optionValue < 0) {
-            event.reply("The " + optionMapping.getName() + " is negative, which isn't supported")
-                .setEphemeral(true)
-                .queue();
-            throw new IllegalArgumentException(optionMapping.getName() + " can't be negative");
-        }
-
-
-        return optionValue;
+    private static OptionalInt getIntegerOptionIfPresent(@Nullable OptionMapping optionMapping) {
+        return Optional.ofNullable(optionMapping)
+            .stream()
```

Comment:
NIT: I dislike the `.stream()` hack. Its a dirty abuse of the API. I know its very common on StackOverflow though... your choice.

```java
return optionMapping == null
  ? OptionalInt.empty();
  : OptionalInt.of(Math.toIntExact(optionMapping.getAsLong()));
```

Also, gets rid of the unecessary boxing conversion from `long` to `Long` that `map(OptionMapping::getAsLong)` introduces...